src/rules.jl

```diff
@@ -161,11 +161,10 @@ hasdiffrule(f::Symbol, arity::Int) = in((f, arity), DEFINED_DIFFRULES)
 @define_diffrule /(x, y) = :(inv($y)),            :(-$x / ($y^2))
 @define_diffrule ^(x, y) = :($y * ($x^($y - 1))), :(($x^$y) * log($x))
 
-# TODO:
-#
-# mod
-# hypot
-# atan2
+@define_diffrule atan2(x, y) = :($y / ($x*$x + $y*$y)), :(-$x / ($x*$x + $y*$y))
```

Comment:
Can we switch `x * x` and `y * y` to `x^2` and `y^2`, just to keep in style with the other rules? Thanks!